pkg/logql/stats/context.go

```diff
@@ -157,8 +164,17 @@ func Snapshot(ctx context.Context, execTime time.Duration) Result {
 		res.Store.CompressedBytes = c.CompressedBytes
 		res.Store.TotalDuplicates = c.TotalDuplicates
 	}
-	res.ComputeSummary(execTime)
-	return res
+
+	// see if there is a pre-computed Result embedded in the context which needs merging
+	_ = JoinResults(ctx, res)
+	merged, err := GetResult(ctx)
+	if err != nil {
```

Comment:
I'll appreciate a log here.

Reply:
ho wait this is a happy path ?

Reply:
The problem I have here is that if you should have pre-computed and you don't then we will never knows. Not sure how you can solve that without being noisy. May be the function should take a parameter for different use case when you expect a pre-computed data or not.